yarn-project/foundation/src/fields/point.ts

```diff
@@ -124,10 +124,10 @@ export class Point {
 
     // If y is null, the x-coordinate is not on the curve
     if (y === null) {
-      throw new NotOnCurveError();
+      throw new NotOnCurveError(x);
     }
 
-    const yPositiveBigInt = y.toBigInt() > (Fr.MODULUS - 1n) / 2n ? Fr.MODULUS - y.toBigInt() : y.toBigInt();
+    const yPositiveBigInt = y.toBigInt() <= (Fr.MODULUS - 1n) / 2n ? y.toBigInt() : Fr.MODULUS - y.toBigInt();
```

Comment:
Sneaked this change in here since it didn't get to previous PR before auto-merge (see [comment](https://github.com/AztecProtocol/aztec-packages/pull/7455#discussion_r1678312112)).